catalogue_graph/src/pit_opener.py

```diff
@@ -0,0 +1,49 @@
+import argparse
+import typing
+
+import config
+from models.events import BasePipelineEvent
+from utils.elasticsearch import ElasticsearchMode, get_client, get_standard_index_name
+
+
+def handler(event: BasePipelineEvent, es_mode: ElasticsearchMode = "private") -> dict:
+    """Create a point in time (PIT) on the denormalised index and return its PIT ID."""
+    es_client = get_client("graph_extractor", event.pipeline_date, es_mode)
+    index_name = get_standard_index_name(
+        config.ES_DENORMALISED_INDEX_NAME, event.pipeline_date
+    )
+    pit = es_client.open_point_in_time(index=index_name, keep_alive="15m")
```

Comment:
The fixed PIT here means steps using it must complete within the 15m window, are we comfortable this will be true even during the reindex case where there will be many thousands of records to process within a window?